src/saturn_engine/core/pipeline.py

```diff
@@ -99,6 +99,7 @@ def instancify_args(cls, args: dict[str, object], pipeline: Callable) -> None:
 class QueuePipeline:
     info: PipelineInfo
     args: dict[str, Any]
+    labels: dict[str, Any] = dataclasses.field(default_factory=dict)
```

Comment:
I don't think labels should be attached to the `QueuePipeline`, they are defined at the `Job` level. I would avoid moving property around just to fit some specfic use case. If we need the Job data in a hook, then let's design the hook to receive job data.

Reply:
Plus I'm not sure why it's required, you can already access the "job" object (`QueueItem`), we just need to add the `labels` there, so we can now do `xmsg.queue.definition.labels`